### DISCUSSION TITLE
Does osgEarth have a plan to support Vulkan?

### DISCUSSION THREAD

**Discussion by yuxiangyu** [pelicanmapping/osgearth#1790] [Support Q&A]
I have seen the vsg project in github (https://vsg-dev.github.io/VulkanSceneGraph/). Weather osgearth has the develop plan to surpport this project?

**Comment by gwaldron** [comment]
The team is doing some R+D work with Vulkan and VSG. However there is no actual plan to support them at this time.

**Reply by christ0phd** [reply]
Any news on this topic?

**Reply by nmielcarek** [reply]
I saw this linked recently in the vsg-users email list which appears to be an interpretation of osgEarth using the VulkanSceneGraph library:

https://github.com/timoore/vsgEarth

**Reply by christ0phd** [reply]
Thanks, I've found it, too. But as far as I understand their approach is something different than porting.

**Comment by timoore** [comment]
On Tue, Aug 23, 2022 at 9:47 AM christ0phd ***@***.***> wrote:

> Thanks, I've found it, too. But as far as I understand their approach is
> something different than porting.
>
>  I would call vsgEarth a "port," or at least the start of one. I took the
approach of using osgEarth layers as data sources for building a stock VSG
scene graph, more or less, instead of diving in with reimplementing the REX
terrain engine and all the other custom osgEarth nodes. No matter the
porting strategy, there will be a lot of work to produce a complete version
of osgEarth that uses Vulkan.

Tim

**Reply by christ0phd** [reply]
Sorry, my fault, so I misunderstood the docs and sources I've found!

**Comment by gwaldron** [comment]
Not a port, but a new library using VSG.
https://github.com/gwaldron/osgearth/discussions/2281
https://github.com/pelicanmapping/rocky
